api/src/services/lottery.service.ts

```diff
@@ -224,4 +257,325 @@ export class LotteryService {
+   */
+  async lotteryExport<QueryParams extends ApplicationCsvQueryParams>(
+    req: ExpressRequest,
+    res: Response,
+    queryParams: QueryParams,
+  ): Promise<StreamableFile> {
+    const user = mapTo(User, req['user']);
+    await this.authorizeLotteryExport(user, queryParams.listingId);
+
+    const workbook = new Excel.Workbook();
+
+    const filename = join(
+      process.cwd(),
+      `src/temp/lottery-listing-${queryParams.listingId}-applications-${
+        user.id
+      }-${new Date().getTime()}.xlsx`,
+    );
+
+    const zipFilePath = join(
+      process.cwd(),
+      `src/temp/lottery-listing-${queryParams.listingId}-applications-${
+        user.id
+      }-${new Date().getTime()}.zip`,
+    );
+
+    await this.createLotterySheet(workbook, {
+      ...queryParams,
+      includeDemographics: true,
+    });
+
+    await workbook.xlsx.writeFile(filename);
+
+    const readStream = createReadStream(filename);
+
+    return new Promise((resolve) => {
+      // Create a writable stream to the zip file
+      const output = fs.createWriteStream(zipFilePath);
+      const archive = archiver('zip', {
+        zlib: { level: 9 },
+      });
+      output.on('close', () => {
+        const zipFile = createReadStream(zipFilePath);
+        resolve(new StreamableFile(zipFile));
+      });
+
+      archive.pipe(output);
+      archive.append(readStream, {
+        name: `lottery-${queryParams.listingId}-${new Date().getTime()}.xlsx`,
+      });
+      archive.finalize();
+    });
+  }
+
+  /**
+   *
+   * @param filename
+   * @param queryParams
+   * @returns generates the lottery sheet
+   */
+  async createLotterySheet<QueryParams extends ApplicationCsvQueryParams>(
+    workbook: Excel.Workbook,
+    queryParams: QueryParams,
+  ): Promise<void> {
+    let applications = await this.prisma.applications.findMany({
+      select: {
+        id: true,
+        preferences: true,
+        householdMember: {
+          select: {
+            id: true,
+          },
+        },
+        applicationLotteryPositions: {
+          select: {
+            ordinal: true,
+            multiselectQuestionId: true,
+          },
+          where: {
+            multiselectQuestionId: null,
+          },
+          orderBy: {
+            ordinal: OrderByEnum.DESC,
+          },
+        },
+      },
+      where: {
+        listingId: queryParams.listingId,
+        deletedAt: null,
+        markedAsDuplicate: false,
+      },
+    });
+
+    // get all multiselect questions for a listing to build csv headers
+    const multiSelectQuestions =
+      await this.multiselectQuestionService.findByListingId(
+        queryParams.listingId,
+      );
+
+    // get maxHouseholdMembers associated to the selected applications
+    let maxHouseholdMembers = 0;
+    applications.forEach((app) => {
+      if (app.householdMember?.length > maxHouseholdMembers) {
+        maxHouseholdMembers = app.householdMember.length;
+      }
+    });
```

Comment:
Assuming that the householdSize field is available you could have this be a reduce function
```
    const maxHouseholdMembers = applications.reduce((maxValue, application) => {
      return Math.max(maxValue, application.householdSize);
    }, 0);